### PR TITLE
rofi-rbw: 1.1.0 -> 1.2.0

### LINKED ISSUE #218311: rofi-rbw has missing runtime dependencies on nixOS 22.11
### Describe the bug
In the nixos-22.11 repo, the package rofi-rbw doesn't run when installed alone.

### Steps To Reproduce
1. Include ``pkgs.rofi-rbw`` in your ``environment.systemPackages``
2. ```nixos-rebuild switch```
3. Try to run ```rofi-rbw```

### Expected behavior
The command would run normally, prompting you to enter your baseurl, username and password.

### Screenshots
image
See https://github.com/fdw/rofimoji/issues/106

If you'd happen to additionally install just ``wtype`` then this would happen:
image

And finally if ``pkgs.rbw`` is available, it would work just fine:
image

### Additional context
Suggested fix:
- Add the following to ``buildInputs``:
```
   buildInputs = [
      rbw
      wtype
   ];
```

### Notify maintainers

@dit7ya 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
nikp123@HP-Box /e/n/s/default> nix-shell -p nix-info --run "nix-info -m"
warning: Nix search path entry 'https://nixos.org/channels/nixos-22.11/nixexpr.tar.xz' cannot be downloaded, ignoring
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/a2mfyh90szmc055ivfnv182b5a4naq90-nix-info
copying path '/nix/store/a2mfyh90szmc055ivfnv182b5a4naq90-nix-info' from 'https://cache.nixos.org'...
warning: Nix search path entry 'https://nixos.org/channels/nixos-22.11/nixexpr.tar.xz' cannot be downloaded, ignoring
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.10-zen2, NixOS, 22.11 (Raccoon), 22.11.20230223.50c23cd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-22.11, nixos-unstable, nixpkgs-22.11"`
warning: Nix search path entry 'https://nixos.org/channels/nixos-22.11/nixexpr.tar.xz' cannot be downloaded, ignoring
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by nim65s**
## Description of changes

This is upgrading rofi-rbw to v1.2.0. 

Upstream changelog is here: https://github.com/fdw/rofi-rbw/releases/tag/1.2.0

I'm also trying to fix #218311, by adding the correct dependencies and arguments for x11 and wayland, through the `rofi-rbw-x11` and `rofi-rbw-wayland` wrappers.

I'm clearly not sure this is the right method, so don't hesitate to boast in the review if needed.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/rofi-rbw/default.nix [hunk 1]
```diff
@@ -1,28 +1,71 @@
...
+, wl-clipboard ? null
+, wtype ? null
+, xclip ? null
+, xdotool ? null
+, rbw }:
+
+assert waylandSupport -> wl-clipboard != null;
+assert waylandSupport -> wtype != null;
+assert x11Support -> xclip != null;
+assert x11Support -> xdotool != null;
```
Comment (ambroisie) [inline pkgs/applications/misc/rofi-rbw/default.nix:18]: I think it's unnecessary to have `null` default values. Also grouping values with their option seems to be the pattern.

```suggestion
, rbw

, waylandSupport ? false
, wl-clipboard
, wtype

, x11Support ? false
, xclip
, xdotool
}:
```

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 248543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>rofi-rbw</li>
    <li>rofi-rbw-wayland</li>
    <li>rofi-rbw-wayland.dist</li>
    <li>rofi-rbw-x11</li>
    <li>rofi-rbw-x11.dist</li>
    <li>rofi-rbw.dist</li>
  </ul>
</details>

File: pkgs/applications/misc/rofi-rbw/default.nix [hunk 2]
```diff
@@ -1,28 +1,69 @@
...
+  ] ++ lib.optionals x11Support [
+    xclip
+    xdotool
+  ];
+
+  wrapper_flags = if waylandSupport
+  then "--typer wtype --clipboarder wl-copy"
+  else if x11Support
+  then "--typer xdotool --clipboarder xclip"
+  else "";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/rofi-rbw/default.nix:63]: Use `lib.optionalString` instead.

P.S.: are X11 and Wayland mutually exclusive?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -33684,6 +33684,14 @@ with pkgs;
 
   rofi-rbw = python3Packages.callPackage ../applications/misc/rofi-rbw { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:33685]: Even if "redundant", please set the `enableX11` and `enableWayland` here.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/rofi-rbw/default.nix (hunk 2 above)
Comment (nim65s) [reply pkgs/applications/misc/rofi-rbw/default.nix:63]: They are not. those options only provide wrappers to start the right tools in the right context.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (nim65s) [reply pkgs/top-level/all-packages.nix:33685]: done

File: pkgs/applications/misc/rofi-rbw/default.nix (hunk 2 above)
Comment (AndersonTorres) [inline pkgs/applications/misc/rofi-rbw/default.nix:63]:
```suggestion
  wrapper_flags =
    lib.optionalString waylandSupport "--typer wtype --clipboarder wl-copy"
    ++ lib.optionalString x11Support "--typer xdotool --clipboarder xclip";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Squash the commits

File: pkgs/applications/misc/rofi-rbw/default.nix [hunk 4]
```diff
@@ -1,28 +1,67 @@
...
+    wl-clipboard
+    wtype
+  ] ++ lib.optionals x11Support [
+    xclip
+    xdotool
+  ];
+
+  wrapper_flags =
+    lib.optionalString waylandSupport "--typer wtype --clipboarder wl-copy"
+    ++ lib.optionalString x11Support "--typer xdotool --clipboarder xclip";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/rofi-rbw/default.nix:61]:
```suggestion
    + lib.optionalString x11Support "--typer xdotool --clipboarder xclip";
```

Oh, the ye olde double plus in strings...

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by equirosa** [comment]
Result of `nixpkgs-review pr 248543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>rofi-rbw</li>
    <li>rofi-rbw-wayland</li>
    <li>rofi-rbw-wayland.dist</li>
    <li>rofi-rbw-x11</li>
    <li>rofi-rbw-x11.dist</li>
    <li>rofi-rbw.dist</li>
  </ul>
</details>

**Review by equirosa — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**